src/Sarif.Driver/Sdk/SkimmerBase.cs

```diff
@@ -16,64 +16,62 @@ public SkimmerBase()
 
         abstract public Uri HelpUri { get;  }
 
-        abstract public string Help { get; }
+        abstract public Message Help { get; }
 
-        private IDictionary<string, string> messageTemplates;
-        private IDictionary<string, string> richMessageTemplates;
+        private IDictionary<string, string> messageStrings;
+        private IDictionary<string, string> richMessageStrings;
 
         abstract protected ResourceManager ResourceManager { get; }
 
-        abstract protected IEnumerable<string> TemplateResourceNames { get; }
+        abstract protected IEnumerable<string> MessageResourceNames { get; }
 
-        virtual protected IEnumerable<string> RichTemplateResourceNames => new List<string>();
+        virtual protected IEnumerable<string> RichMessageResourceNames => new List<string>();
 
         virtual public RuleConfiguration Configuration {  get { return RuleConfiguration.Enabled; } }
 
         virtual public ResultLevel DefaultLevel { get { return ResultLevel.Warning; } }
 
-        virtual public IDictionary<string, string> MessageTemplates
+        virtual public IDictionary<string, string> MessageStrings
         {
             get
             {
-                if (this.messageTemplates == null)
+                if (this.messageStrings == null)
                 {
-                    this.messageTemplates = InitializeMessageTemplates();
+                    this.messageStrings = InitializeMessageTemplates();
```

Comment:
`InitializeMessageTemplates` => `InitializeMessageStrings`.

Search "template" case-insensitive throughout the code base to catch these stragglers.

Reply:
Yah meant to do this...

Reply:
And done.